core/src/test/java/org/apache/iceberg/TestMergeAppend.java

```diff
@@ -179,10 +180,14 @@ public void testManifestMergeMinCount() throws IOException {
     Assert.assertEquals("Should contain 2 merged manifest for first write",
         2, readMetadata().currentSnapshot().manifests().size());
 
+    ManifestFile manifest4 = writeManifestWithName("FILE_A", FILE_A);
+    ManifestFile manifest5 = writeManifestWithName("FILE_C", FILE_C);
+    ManifestFile manifest6 = writeManifestWithName("FILE_D", FILE_D);
+
     table.newAppend()
-        .appendManifest(manifest)
-        .appendManifest(manifest2)
-        .appendManifest(manifest3)
+        .appendManifest(manifest4)
```

Comment:
This change is needed because some of the manifests will be merged during the previous append and they will be deleted automatically.